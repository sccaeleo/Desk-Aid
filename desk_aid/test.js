
/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';
import axios from 'axios';
import '@testing-library/jest-dom';

// Page Imports
import Home from './src/app/page.js';
//import GuideList from './src/app/GuideListPage/page.js';
//import Guide from './src/app/GuidePage/page.js';
import Resources from './src/app/ResourcesPage/page.js';
//import Help from './src/app/HelpPage/page.js';
//import SignIn from './src/app/SignInPage/page.js';

jest.mock('axios');

// Home Page Tests
describe('Home page', () => {
  axios.get.mockResolvedValue({
    data: [
      { name: 'Category 1' },
      { name: 'Category 2' },
      { name: 'Category 3' },
    ],
  });
  it('Displays title', () => {
    const { getByText } = render(<Home />);
    expect(getByText('Constituent Service Desk Aid')).toBeInTheDocument();
  });

  it('Renders search bar', () => {
    const { getByPlaceholderText } = render(<Home />);
    expect(getByPlaceholderText('Search')).toBeInTheDocument();
  });

  it('Calls search function', () => {
    const searchMock = jest.fn();
    const { getByText, getByPlaceholderText } = render(<Home search={searchMock} />);
    const searchInput = getByPlaceholderText('Search');
    const submitButton = getByText('Search');
    fireEvent.change(searchInput, { target: { value: 'test query' } });
    fireEvent.click(submitButton);
    waitFor(() => expect(searchMock).toHaveBeenCalledTimes(1));
  });
});

describe('Resources page', () => {
  it('Renders resources', async () => {
    const resources = [
      { name: 'Resource 1' },
      { name: 'Resource 2' },
      { name: 'Resource 3' },
    ];

    axios.get.mockResolvedValue({ data: resources });

    const { getByText } = render(<Resources />);
    await waitFor(() => expect(getByText('Resource 1')).toBeInTheDocument());
    await waitFor(() => expect(getByText('Resource 2')).toBeInTheDocument());
    await waitFor(() => expect(getByText('Resource 3')).toBeInTheDocument());
  });

});