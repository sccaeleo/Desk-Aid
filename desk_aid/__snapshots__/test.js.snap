// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page Renders 1`] = `
<div>
  <div
    class="flex-col text-center"
  >
    <h1
      class="text-5xl font-bold pt-20"
    >
      Constituent Service Desk Aid
    </h1>
    <form
      style="position: absolute; top: 40vh; width: 100%;"
    >
      <input
        class="text-black w-4/5 rounded-l-md"
        placeholder="Search"
        type="text"
        value=""
      />
      <button
        class="hover:bg-blue-500 rounded-r-md px-4"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</div>
`;

exports[`Home page renders correctly 1`] = `
<div>
  <div
    class="flex-col text-center"
  >
    <h1
      class="text-5xl font-bold pt-20"
    >
      Constituent Service Desk Aid
    </h1>
    <form
      style="position: absolute; top: 40vh; width: 100%;"
    >
      <input
        class="text-black w-4/5 rounded-l-md"
        placeholder="Search"
        type="text"
        value=""
      />
      <button
        class="hover:bg-blue-500 rounded-r-md px-4"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</div>
`;

exports[`renders correctly 1`] = `null`;
